src/caffe/layers/base_data_layer.cpp

```diff
@@ -77,7 +77,7 @@ void BasePrefetchingDataLayer<Dtype>::InternalThreadEntry() {
 #ifndef CPU_ONLY
   cudaStream_t stream;
   if (Caffe::mode() == Caffe::GPU) {
-    CUDA_CHECK(cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking));
+    CUDA_CHECK(cudaStreamCreate(&stream));
```

Comment:
Not sure anymore.  It's in the prefetching layer, so we might be relying on blocking streams for synchronization since that stream is limited I believe to the data layer threads moving data to the GPU.